Doc/whatsnew/3.7.rst

```diff
@@ -483,138 +782,458 @@ The :func:`~binascii.b2a_uu` function now accepts an optional *backtick*
+The new methods return tuples and are not restricted by the range supported by
+:class:`datetime.date`.
+(Contributed by Alexander Belopolsky in :issue:`28292`.)
+
+
+collections
+-----------
+
+The :func:`collections.namedtuple` factory has been reimplemented to rely
+on precomputed methods to substantially reduce the use of :func:`exec`.
+As a result, the *verbose* parameter and the *_source* attribute of the
+named tuple are not longer supported.  The benefits include:
+1) having a smaller memory footprint for applications using multiple named
+tuples, 2) faster creation of the named tuple class (approx 4x to 6x depending
+on how it is measured), and 3) minor speed-ups for instance creation.
+(Jelle Zijlstra with further improvements by INADA Naoki, Serhiy Storchaka,
+and Raymond Hettinger in :issue:`28638`.)
+
+Additionally, ``collections.namedtuple()`` now supports default values.
+(Contributed by Raymond Hettinger in :issue:`32320`.)
+
+:class:`collections.ChainMap` now preserves the order of the underlying
+mappings.  (Contributed by Raymond Hettinger in :issue:`32792`.)
+
+
+compileall
+----------
+
+:func:`compileall.compile_dir` learned the new *invalidation_mode* parameter,
+which can be used to enable
+:ref:`hash-based .pyc invalidation <whatsnew37-pep552>`.  The invalidation
+mode can also be specified on the command line using the new
+``--invalidation-mode`` argument.
+(Contributed by Benjamin Peterson in :issue:`31650`.)
+
+
+concurrent.futures
+------------------
+
+:class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>` and
+:class:`ThreadPoolExecutor <concurrent.futures.ThreadPoolExecutor>` now
+support the new *initializer* and *initargs* constructor arguments.
+(Contributed by Antoine Pitrou in :issue:`21423`.)
+
+The :class:`ProcessPoolExecutor <concurrent.futures.ProcessPoolExecutor>`
+can now take the multiprocessing context via the new *mp_context* argument.
+(Contributed by Thomas Moreau in :issue:`31540`.)
+
+``executor.submit()`` now raises a :exc:`RuntimeError` if called during
+interpreter shutdown.
+(Contributed by Mark Nemec in :issue:`33097`.)
+
+
+configparser
+------------
+
+The :class:`~configparser.ConfigParser`` constructor now uses ``read_dict()``
+to process the default values, making its behavior consistent with the
+rest of the parser.  Non-string keys and values in the defaults
+dictionary are now being implicitly converted to strings.
+(Contributed by James Tocknell. in :issue:`23835`.)
+
+The *filename* parameter of the
+:meth:`ConfigParser.read() <configparser.ConfigParser.read>` method now
+accepts bytes objects.
+(Contributed by Vincent Michel. in :issue:`31307`.)
+
+
 contextlib
 ----------
 
-:func:`~contextlib.asynccontextmanager` and
-:class:`~contextlib.AbstractAsyncContextManager` have been added. (Contributed
+The new :func:`~contextlib.nullcontext` is a simpler and faster alternative
+to :class:`~contextlib.ExitStack` for handling simple context managers.
```

Comment:
I don't think that "handling simple context managers" is an accurate description of what `nullcontext` is good for. It won't help you handle or implement a simple context manager, it's useful only as a no-op context managers. Suggested re-word: "... a simpler and faster no-op context manager than ExitStack."